function-server/src/test/java/io/dispatchframework/javabaseimage/ServerTest.java

```diff
@@ -0,0 +1,81 @@
+package io.dispatchframework.javabaseimage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Type;
+import java.util.function.BiFunction;
+
+public class ServerTest {
+    @Test
+    public void getBiFunctionTypes_WithInterface() throws ClassNotFoundException {
+        final String packageName = "io.dispatchframework.javabaseimage";
+        final String className = "GoodHandler";
+        Server.ExecFunction execFunction = new Server.ExecFunction(packageName, className);
+        Class<?> functionClass = Class.forName(packageName + "." + className);
+        Type[] biFunctionTypes = execFunction.getBiFunctionTypes(functionClass);
+
+        assertEquals(3, biFunctionTypes.length);
+        assertTrue(biFunctionTypes[0].getTypeName().equals("java.util.Map<java.lang.String, java.lang.Object>"));
+        assertTrue(biFunctionTypes[1].getTypeName().equals("java.util.List<java.lang.String>"));
+        assertTrue(biFunctionTypes[2].getTypeName().equals("java.util.Set<java.lang.Integer>"));
+    }
+
+    @Test
+    public void getBiFunctionTypes_WithoutInterface() throws ClassNotFoundException {
```

Comment:
I'd test that if the function implementation is not good, a runtime exception is thrown from `getBiFunctionTypes()`.

Then, if we move the call to `getBiFunctionTypes()` into the constructor (as commented above), it prevents us from thinking that we have a good working function awaiting invocations.